Classes/LinkHandler/PageLinkHandler.php

```diff
@@ -147,17 +159,36 @@ protected function getRecordsOnExpandedPage($pageId)
                     )
                 )
                 ->orderBy('colPos')
-                ->addOrderBy('sorting')
-                ->execute()
-                ->fetchAll();
+                ->addOrderBy('sorting');
+
+            $versionInformation = GeneralUtility::makeInstance(Typo3Version::class);
+            if($versionInformation->getMajorVersion() < 12) {
+                $contentElements = $contentElements
```

Comment:
Personally, I don't like to "hijack" the result variable name from the query builder and assign it anew, which is then an array. I would prefer to use a different variable name. Also some lines below.